test/extensions/filters/network/http_connection_manager/config_test.cc

```diff
@@ -36,7 +36,6 @@ parseHttpConnectionManagerFromV2Yaml(const std::string& yaml) {
   return http_connection_manager;
 }
 
-// TODO(yittg): always validate config and split all cases using deprecated feature.
```

Comment:
Can you remove `parseHttpConnectionManagerFromV2Yaml` now above? Also, it might be a simpler change to just move the validate call to the function above vs. change the function called everywhere, but up to you.